### DISCUSSION TITLE
UX: Logs: Show only part of long multi-line log message

### DISCUSSION THREAD

**Discussion by adambro** [grafana/grafana#57415] [Observability]
### Background / Why we're doing this

Some log evens can contain hundreds of lines, typical for Java stack traces. When we've deployed the `multiline` filter to concatenate logs lines for single exception stack trace it is displayed in full on the Logs panel inside Explore interface.

Right now having such hundreds lines in log message it is impossible to quickly scan the logs to spot some unusual patterns in the explore view.

### Is there existing research for this?

No research yet

### Problems or tasks

Grafana could limit the number of characters from the message, or display just the first line, in the  list view.

For log entry details view I'd like to see the multi-line stack trace in full. but only after clicking to see the details.

### Deliverables

The `LogRowMessage` component changes I guess.

**Comment by adambro** [comment]
Maybe it can use `wrapLogMessage` value to show only first line when wrap message is false? Just an idea for quick win.

**Comment by adambro** [comment]
Screenshot with example in this tweet https://twitter.com/AdamBrodziak/status/1583073367179276288

**Comment by adambro** [comment]
Also there are empty spaces rendered in Grafana Loki in explore view. I think they are caused by styles applied in our light theme, specifically `line-height: 1.5;` CSS rule that is applied to <body> elem. See the screenshot.
https://twitter.com/AdamBrodziak/status/1583420792489902080

Maybe row log message CSS could reset line-height?

**Comment by ivanahuckova** [comment]
Hello @adambro and thank you for feature request. We keep all the feature request in discussion, so I am going to transfer this to discussion. 

Regarding your comment about spacing, I agree that spacing seems off. This seems like a UI issue, so I am going to create new issue from your comment. For future cases, it is better to keep each issue separately.

**Comment by fdrouet** [comment]
Any news on this painful usability issue ?

**Comment by steven-sheehy** [comment]
I would like to see this as well. I now leverage promtail's `multiline` stage to properly capture my multiline Java stacktrace as a single message. This makes it easier to use with logql and view individual messages, but it makes it harder to visually view multiple messages when they contain multiple lines. I agree with @adambro and think it should default to only showing the first row and expand additional lines when clicked.

**Comment by henrique1972** [comment]
Any news on this painful usability issue ?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82970. Feel free to continue the discussion around this there. Thank you!

**Comment by adambro** [comment]
Your could just reopen #56893 instead. That was my original feature request.

**Reply by gelicia** [reply]
@adambro looks like I cant actually re-open that issue :( Sorry for the back and forth.
